### DISCUSSION TITLE
follow_mouse

### DISCUSSION THREAD

**Discussion by aloispichler** [hyprwm/Hyprland#7677] [Q&A]
I have `follow_mouse= 2` and the behavior is as in the docs, “2 - Cursor focus will be detached from keyboard focus. Clicking on a window will move keyboard focus to that window”.

However, this is not the case if there is a floating window (in my case, `windowrule = float, ^(xpad)$`). Hovering the window with the mouse _does_ change the cursor focus.

The same with `follow_mouse= 0`, although the docs say “0 - Cursor movement will not change focus”.

Did I miss an option to have the same behavior for all windows, irrespective if floating or not?

Thanks

**Comment by dterhorst** [comment]
## Hyprland version
❯ hyprctl version
Hyprland 0.48.1 built from branch  at commit 29e2e59fdbab8ed2cc23a20e3c6043d5decb5cdc  (version: bump to v0.48.1).
Date: Fri Mar 28 16:16:07 2025
Tag: v0.48.1, commits: 5937
built against:
 aquamarine 0.8.0
 hyprlang 0.6.0
 hyprutils 0.6.0
 hyprcursor 0.1.12
 hyprgraphics 0.1.3


no flags were set

## Regression
As a relatively new user, I am not able to judge if this is a regression.

## Describe the bug
Floating windows such as "Save As..." dialogs do not seem to respect `follow_mouse = 2`.
Moving the mouse causes other windows to be focused without clicking them.

## Reproduction steps
1. Set the following property in `hyprland.conf`:
```
input {
    ...
    follow_mouse = 2
    ...
}
```
2. Install and open Krita.
3. Create a new document.
4. Press "Ctrl+S" to save. This should trigger a "Save As..." dialog.
5. Move the mouse so that it is above the parent Krita window instead of the "Save As..." dialog.

Expected:
The "Save As..." dialog should remain focused.

Actual:
The parent Krita window becomes focused.

## System info and config
[systeminfo.txt](https://github.com/user-attachments/files/19722448/systeminfo.txt)
[clients.txt](https://github.com/user-attachments/files/19722449/clients.txt)

## Notes
Searching for existing issues brought up this post. However, OP did not follow the posting guidelines when creating this discussion. I have attempted to rectify that by providing the missing information, but there are some things I can't fix. @aloispichler, please change the Category from "Q&A" to "Config" and/or "Input". Thank you.

**Comment by ghost** [comment]
according to vaxry floating windows and tiled windows should act separately (or something), if there are all floating windows on a workspace and no tiled windows then follow_mouse = 2 will work properly, same with tiled windows but when you mix the 2 it will only work for 1 type

**Comment by dterhorst** [comment]
It appears to me that you have described a technical limitation which contributes to the problem...
...but you haven't described a reason why the problem shouldn't be fixed.

Even if the layers are kept separate, wouldn't it be good to have a policy for how to switch between those layers? A policy like `follow_mouse`...?

All I know is that I've made every effort to disable focus-follows-mouse and I have run out of options.

**Reply by ghost** [reply]
https://github.com/hyprwm/Hyprland/pull/9994 somethin like this mght help

**Reply by dterhorst** [reply]
If I understand correctly, that would allow a rule like this?

```
windowrulev2 = nofollowmouse, class:^(krita)$, title:^(Saving As — Krita)$
```

**Reply by ghost** [reply]
yes

**Reply by dterhorst** [reply]
If that works, it would resolve my problem. Thank you.

**Comment by MightyPlaza** [comment]
`float_switch_override_focus = 0` ?

**Reply by dterhorst** [reply]
Oh, nice. That broadly works. Thank you.

Krita has a problem with it:

> If I open Krita and try to save a document, the "Save As..." dialog appears.
Then, if I click on the parent Krita window, **my mouse disappears**.
The mouse reappears if I move it completely outside of the parent Krita window, but disappears again if I move it inside.
The mouse remains gone even if I manage to blindly position it so that it's over the "Save As..." dialog.
(EDIT: If the "Save As..." dialog is positioned halfway over Krita and halfway over another window, the cursor will appear normally if it is positioned over the part of the "Save As..." dialog that overlaps the non-Krita window.)
But if I manage to blindly click the "Save As..." dialog, the mouse re-appears.
But only while the cursor is above the "Save As..." dialog.
If I hover over the parent Krita window, it disappears again. And re-appears if I hover over "Save As..."

But it seems to be a Krita-only problem as far as I can tell.
